lib/ruby_lsp/requests/support/dependency_detector.rb

```diff
@@ -38,7 +38,8 @@ def detected_test_library
 
       sig { params(gem_pattern: Regexp).returns(T::Boolean) }
       def direct_dependency?(gem_pattern)
-        Bundler.locked_gems.dependencies.keys.grep(gem_pattern).any?
+        File.exist?("#{Dir.pwd}/Gemfile") &&
+          Bundler.locked_gems.dependencies.keys.grep(gem_pattern).any?
```

Comment:
This does not consider the `BUNDLE_GEMFILE` env variable nor the fact that the `Gemfile` can be in a parent folder. I think we should rather use the normal `Bundler` methods for this:
```suggestion
        Bundler.default_gemfile &&
          Bundler.locked_gems.dependencies.keys.grep(gem_pattern).any?
      rescue Bundler::GemfileNotFound # Bundler.default_gemfile raises this if no Gemfile found
        nil
```

Reply:
This won't work because it always returns `BUNDLE_GEMFILE` if it is set ([here](https://github.com/rubygems/rubygems/blob/e2e2e7cff176ad0d25fa6e9c0d21a500217b47bd/bundler/lib/bundler/shared_helpers.rb#L215-L216)).

It's always going to be set because the LSP will either point to the custom bundle or the main bundle if the gem is in the `Gemfile`, so there's no scenario where this will raise.

Reply:
I am not sure if I understand this. If the Ruby LSP gem is in the `Gemfile`, then we don't want `Bundler.default_gemfile` to fail, since we know that the project uses a `Gemfile`. So it works in that case.

For the case that Ruby LSP is working against the custom bundle, do we not need the custom bundle to be sourcing the original Gemfile as well? In any case, maybe we could be doing all this in a `Bundler.with_original_env` block if we don't want the custom bundle targeting to be a problem.

Reply:
This fix is for when there is no top level `Gemfile`. So, yeah, `Bundler.with_original_env` does the trick in this case since we want to check the `default_gemfile` ignoring the custom bundle.